encoding/descaling.md

```diff
@@ -0,0 +1,452 @@
+descale_c = core.descale.Debicubic(Y, width, height, b=0, c=1).resize.Bicubic(1920, 1080, filter_param_a=0, filter_param_b=1)            # Sharp Bicubic
+descale_d = core.descale.Debicubic(Y, width, height, b=1, c=0).resize.Bicubic(1920, 1080, filter_param_a=1, filter_param_b=0)            # B-Spline
+descale_e = core.descale.Debicubic(Y, width, height, b=0, c=1/2).resize.Bicubic(1920, 1080, filter_param_a=0, filter_param_b=1/2)        # Catmull-rom
+descale_f = core.descale.Despline36(Y, width, height).resize.Spline36(1920, 1080)                                                        # Spline36
+
+descaled = kgf.join([descale_a, U, V])
+descaled.set_output()
+```
+
+You might realize that after descaling,
+we are immediately upscaling the frame
+with the same kernel and values again.
+This is done so we can compare the before and after.
+The closer the new frame is to the old one,
+how more likely it is that you've got the correct kernel.
+Zooming in on the frame at 4x magnification or higher
+will help immensely.
+An alternative that you can do
+is to simply descale until
+you've got what you believe to be the best result.
+It's faster to do it this way,
+but might be less accurate.
+
+[manaria_compare]: https://slowpics.org/comparison/61e39e1e-d074-4d83-b7c9-b0f4e1861855
+
+
+## Credits and other native 1080p elements
+
+There is one very, very important thing
+to keep in mind when descaling:
+
+*Credits are usually done in 1080p*.
+
+There are various masks you can use
+to help with dealing with that issue,
+but it might be better
+to make use of existing wrappers instead.
+For this example I'll
+be using `inverse_scale` from kagefunc.
+
+```Py
+descaled = kgf.inverse_scale(src, height=878, kernel='bicubic', b=0, c=1/2, mask_detail=True)
+```
+
+We can make use of the mask
+that `inverse_scale` uses internally
+as well.
+
+```Py
+descaled = kgf.inverse_scale(src, height=874, kernel='bicubic', b=0, c=1/2)
+descaled_mask = kgf._generate_descale_mask(vsutil.get_y(core.resize.Spline36(src, descaled.width, descaled.height)), vsutil.get_y(descaled), kernel='bicubic', b=0, c=1/2)
+```
+
+![KaguyaOP_credits_mask](images/descale_credits_mask.png)
+![KaguyaOP_descaled](images/descale_credits.png)
+
+Note that if you see the mask
+catching a lot of other stuff,
+you might want to consider *not* descaling
+that particular frame,
+or to try a different kernel/values.
+Chances are that you're either
+using the wrong kernel
+or that the frames you're looking at are native 1080p.
+
+![native1080.png](images/descale_native1080.png)
+
+![dontdescalethis.png](images/descale_dontdescalethis.png)
+
+
+## Dealing with bad descaling
+
+There are various things you can do
+to deal with scenes that have issues
+even after descaling.
+`Eedi3` stands out in particular
+as a fantastic AA filter
+that really nails down bad lineart
+caused by bad descaling.
+Just try not to rely on it too much,
+as it is by all means a "cheat code".
```

Comment:
Don’t make it sounds so good or people will think it magically fixes everything. Like descaling to the wrong resolution and then having the detail mask and eedi carry you. :smug_nep: